src/backend/distributed/sql/udfs/citus_isolation_test_session_is_blocked/latest.sql

```diff
@@ -23,10 +23,24 @@ RETURNS boolean AS $$
         FROM get_all_active_transactions() WHERE process_id = pBlockedPid;
     END IF;
 
-    RETURN EXISTS (
-      SELECT 1 FROM citus_internal_global_blocked_processes()
-        WHERE waiting_global_pid = mBlockedGlobalPid
-    );
+    -- normally the "off" version should be enough, but some isolation tests
+    -- expose problems in our gpid processing that cause some blocks to be
+    -- missed. For those we explicitly enable checking for all blocks, which
+    -- has some risks of flakyness.
+    IF current_setting('citus.isolation_test_check_all_blocks') = 'off' THEN
+      RETURN EXISTS (
+        SELECT 1 FROM citus_internal_global_blocked_processes()
+          WHERE waiting_global_pid = mBlockedGlobalPid
+          AND blocking_global_pid in (
+              SELECT citus_calculate_gpid(NULL, pid) FROM unnest(pInterestingPids) pid
```

Comment:
I don't want to drag this PR a lot, but still I don't think this trick would solve the problem of us using coordinator local processIds+nodeId (e.g., pInterestingPids) with `blocking_global_pid`. 

`blocking_global_pid` could be a gpid initiated from the worker, and we are still using coordinator's nodeId. That is useful for commands that are blocked on the coordinator.

I think to have this filter on `blocking_global_pid`, we should probably calculate `pInterestingGPids` from all nodes. Maybe we do:
```SQL
AND blocking_global_pid in (
              SELECT blocking_global_pid FROM citus_internal_global_blocked_processes())